src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1398,6 +1423,61 @@ static bool isLessDerived(TypeSymbol type, NamedTypeSymbol currentType, ref Comp
             }
         }
 
+        private static bool IsLessSpecificThanAnyExtensionMember<TMember>(int index, TypeSymbol extendedType, ArrayBuilder<MemberResolutionResult<TMember>> results, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+            where TMember : Symbol
+        {
+            for (int j = 0; j < results.Count; ++j)
+            {
+                if (j == index)
+                {
+                    continue;
+                }
+
+                var result = results[j];
+
+                if (!result.Result.IsValid)
+                {
+                    continue;
+                }
+
+                if (result.Member.ContainingType.ExtendedTypeNoUseSiteDiagnostics is not { } currentExtendedType)
+                {
+                    continue;
+                }
+
+                if (isExtendedTypeLessDerived(extendedType, currentExtendedType, ref useSiteInfo))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+
+            static bool isExtendedTypeLessDerived(TypeSymbol extendedType, TypeSymbol type, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
```

Comment:
>isExtendedTypeLessDerived [](http://example.com/codeflow?start=24&length=25)

Is this pretty much a dupe of Binder.IsDerivedType from Binder_Lookup.cs? If so, consider calling that helper instead.